warzone/Player.cpp

```diff
@@ -1 +1,110 @@
 #include "Player.h"
+
+Player::Player() {  //default constructor
+	this->playerTerritories = new vector<Territory*>();
+	this->playerHand = new Cards::Hand();
+	this->playerOrdersList = new OrdersList();
+}
+
+
+Player::Player(vector<Territory*>* territoriesToAdd, OrdersList* playerList, Cards::Deck* deckToTakeFrom) { //Constructor with all parameters
+	this->playerTerritories = territoriesToAdd;
+	this->playerOrdersList = playerList;
+	this->playerHand = new Cards::Hand(deckToTakeFrom);
+}
+
+Player::~Player() {  //destructor 
+	for (Territory* t : *playerTerritories) {
+		delete t;
+	}
+	delete playerTerritories;
+	delete playerHand;
+	delete playerOrdersList;
+}
+
+Player& Player::operator=(const Player& one)  //assignment operator
+{
+	delete playerHand;
+	this->playerHand = one.playerHand;
+	delete playerTerritories;
+	this->playerTerritories = one.playerTerritories;
+	delete playerOrdersList;
+	this->playerOrdersList = one.playerOrdersList;
+	return *this;
```

Comment:
I don't think this implementation is correct, you can just use the copy constructor
`return new Player(one);`

Reply:
sounds good!